### ISSUE TITLE
Core package implementation bug

### ISSUE THREAD

**Issue by oliv0533** [MarkusPalcer/blurhash.net#8]
The BlurHash.Core package CoreEncoder might have a bug. The CoreEncode method is protected. Every project using this package need to inherit this class somewhere to call this method. This seems like a bug, since this isn't mentioned in the documentation. The same is true for The CoreDecoder class and its CoreDecode method.

**Comment by MarkusPalcer** [comment]
This is intended.
The Core* methods are the _internal_ methods which are oblivious to how an image is represented on the outside world.
As seen in the Blurhash-System.Drawing, it is _meant_ to be inherited to an Encoder which is specific to a concrete image representation (System.Drawing.Image in that case).

The Core*-Methods _must not be used_ directly.

**Comment by oliv0533** [comment]
In that case, i would update your documentation. 

The documentation implies that you can instantiate the Core Encoder/Decoder and call the method. I would reword that, so you implement a decoder or encoder and then instantiate that implementation. 

The architecture makes sense, but the documentation implies a different intent, hence the confusion on my part.

**Comment by MarkusPalcer** [comment]
Yes, I will definitely do that :)
